map_model/src/edits/mod.rs

```diff
@@ -454,6 +451,73 @@ fn recalculate_turns(id: IntersectionID, map: &mut Map, effects: &mut EditEffect
     }
 }
 
+fn modify_lanes(
+    map: &mut Map,
+    r: RoadID,
+    lanes_ltr: Vec<(LaneType, Direction)>,
+    effects: &mut EditEffects,
+) {
+    let road = &mut map.roads[r.0];
+
+    // TODO Widening roads is still experimental. If we're just modifying lane types, preserve
+    // LaneIDs.
+    if road.lanes_ltr.len() == lanes_ltr.len() {
```

Comment:
This is the old logic, copied over. I think it can eventually get deleted, once we replace `LaneEditor` with Yuwen's new design. But in the meantime, we don't want LaneIDs to change.